src/DeploymentTemplate.ts

```diff
@@ -228,21 +243,37 @@ export class DeploymentTemplate extends DeploymentDocument {
         return warnings;
     }
 
-    private findUnusedUserFunctions(): language.Issue[] {
+    /**
+     * Finds parameters/variables/functions inside outer-scoped nested templates, which
+     * by definition can't be accessed in any expressions.
+     */
+    private findInaccessibleScopeMembers(): language.Issue[] {
         const warnings: language.Issue[] = [];
+        const warningMessage =
+            // tslint:disable-next-line: prefer-template
+            'Variables, parameters and user functions of an outer-scoped nested template are inaccessible to any expressions. '
+            + `If you intended inner scope, set properties.nestedDeploymentExprEvalOptions.scope to 'inner'.`;
+
+        for (const scope of this.allScopes.filter(ts => ts.scopeKind === TemplateScopeKind.NestedDeploymentWithOuterScope)) {
+            const parameters = scope.rootObject?.getProperty(templateKeys.parameters)?.value;
```

Comment:
>scope.rootObject?.getProperty(templateKeys.parameters)?.value [](start = 31, length = 61)

Nit: I personally would've used a helper function for this fairly complicated expression since it's repeated three times with a different property name. Same for !!parameters?.xxxx bool expression below. Actually, it looks like even a large helper could've been created that parameterized 

             const parameters = scope.rootObject?.getProperty(templateKeys.parameters)?.value;
            // tslint:disable-next-line: strict-boolean-expressions
            if (!!parameters?.asObjectValue?.properties?.length) {
                warnings.push(
                    new language.Issue(parameters.span, warningMessage, language.IssueKind.inaccessibleNestedScopeMembers));
            }

Reply:
Refactored what I could.  The difference between objects and arrays made it limited.